### PR TITLE
Switch to `hatch` for pypi publishing

### PR DESCRIPTION
**Pull request by williamshen-nz**
- Make README look nicer
- Add dedicated CONTRIBUTING readme
- Switch to `pyproject.toml` with hatch for build/releasing

So we can `pip install lisdf` now that more packages depend on it.

Need to test on predicators

### CODE REVIEW

**Comment by tomsilver** [comment]
love the logos 😍 looks like they're not displaying properly though: https://github.com/Learning-and-Intelligent-Systems/lisdf/tree/hatch-build

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+* The 2nd command is the linter check, which runs `flake8`
+* The 3rd command is the static typing check, which uses `mypy` to verify type annotations. 
+* The 4th command is the unit testing check, which verifies that unit tests pass and that code is adequately covered. 
+  * The "75" means that 75% of all lines in every file in `lisdf/` must be covered (excludes `tests/`).
+
+### Publishing to PyPI
+Right now, the plan is to manually publish to PyPI when required. We can automate the release process later if needed.
+
+We use [Hatch](https://hatch.pypa.io/latest/) as our build system because it is lightweight, easy to use yet powerful
+in comparison to other tools like `poetry`.
```
Comment (tomsilver) [inline CONTRIBUTING.md:36]: no need to take a shot at poetry here 😉

**Comment by williamshen-nz** [comment]
> love the logos 😍 looks like they're not displaying properly though: https://github.com/Learning-and-Intelligent-Systems/lisdf/tree/hatch-build

Thanks! The URL pointed to the main branch so once this is merged it will show up

**Review by tomsilver — APPROVED** [review]
awesome stuff!

File: README.md [hunk 2]
```diff
@@ -75,9 +76,33 @@ Follow the instructions below:
 4. Check [CONTRIBUTING.md](CONTRIBUTING.md) for more information on how to contribute to this repository, including 
     how to run the tests, code checks, and publishing to PyPI.
 
+
 ## License
 This repository is licensed under the MIT License. See [LICENSE](LICENSE) for more details.
 
+
+## Authors
+LISdf is an intitiative within the [Learning and Intelligent Systems (LIS) group](lis.csail.mit.edu) at
```
Comment (tomsilver) [inline README.md:85]: typo: initiative

**Review by NishanthJKumar — APPROVED** [review]
Fantastic work! Very excited that this is going to be on PyPI (and that it has a spiffy logo) :grin:

**Pull request merged**